script/upload_to_aptly

```diff
@@ -12,31 +12,41 @@ set -exu
 # DOESN'T HAVE A SIGNING SUBKEY.
 #
 # Date          Channel       Fingerprint           Full Key
-#
 # 06/26/2019    Release       0x0BB75829C2D4E821    D8BAD4DE7EE17AF52A834B2D0BB75829C2D4E821
 # 12/29/2018    Beta          0x0B31DBA06A8A26F9    9228DBCE20DDE5EC46488DE90B31DBA06A8A26F9
 # 12/29/2018    Dev           0x0B31DBA06A8A26F9    9228DBCE20DDE5EC46488DE90B31DBA06A8A26F9
 # 12/29/2018    Nightly       0x0B31DBA06A8A26F9    9228DBCE20DDE5EC46488DE90B31DBA06A8A26F9
 
 ### The first argument is the S3 bucket, the second is the GPG key fingerprint
-OPTION=${1:-brave-browser-apt-staging-nightly}
-GPG_KEY_ID=${2:-D8BAD4DE7EE17AF52A834B2D0BB75829C2D4E821} # Defaults to New Brave Software signing key
+S3_BUCKET=${1}
+GPG_KEY_ID=${2}
+DIST='artful zesty yakkety xenial jessie trusty serena stretch bionic buster cosmic disco eoan bullseye stable'
+# SNAP_TAG=$(date +'%s')
 
-if [[ $2 == 0x* ]]; then
-    echo "Error: Please pass the Full GPG KEY ID to sign with! (i.e. '9228DBCE20DDE5EC46488DE90B31DBA06A8A26F9')"
+if [ -n "$S3_BUCKET" ]; then
+    echo "Error: Please pass the S3 bucket argument as the first argument"
     exit 1
 fi
 
-if [ "$GPG_KEY_ID" = "D8BAD4DE7EE17AF52A834B2D0BB75829C2D4E821" ]; then
-  KEY_NAME=brave-core.asc
-else
-  KEY_NAME=brave-core-nightly.asc
+if [ -n "$GPG_KEY_ID" ]; then
+    echo "Error: Please pass the full GPG Key ID as the second argument"
+    exit 1
 fi
-KEY_FILE=/tmp/${KEY_NAME}
-SNAP_TAG=`date +'%s'`
-DIST='artful zesty yakkety xenial jessie trusty serena stretch bionic buster cosmic disco bullseye'
 
-cat <<EOF > $HOME/.aptly.conf
+case "$GPG_KEY_ID" in 
+    "D8BAD4DE7EE17AF52A834B2D0BB75829C2D4E821"|"9BD198A2D848C482E3550697A8580BDC82D3DC6C")  # Release keys
```

Comment:
The first one is the master key and the second one is the "next" subkey. Shouldn't this list both subkeys?

Reply:
Previously the master key was being used to sign the apt repositories, so I believe the master key is being passed in. As part of the switch to the new key I'd like to try using a subkey for signing and distribution.

Reply:
Ah, I thought you were going to try to restore the keyring with both subkeys as part of landing this.

If that's not the case (we can do that in a separate step), then let's keep the master key in there, but also add the old/current subkey too so that we don't have to edit this script to test the keyring with two subkeys.

Reply:
I'm hoping to have the new subkey added to the same master key as part of the keychain. I believe I've fixed the mis-signing by adding a `!` suffix and checked that `aptly` and `rpmsign` sign with the correct subkey.

Reply:
Fixed by adding the old 2048-bit subkey to the case statement.

Reply:
> I'm hoping to have the new subkey added to the same master key as part of the keychain. I believe I've fixed the mis-signing by adding a `!` suffix and checked that `aptly` and `rpmsign` sign with the correct subkey.

I think you could test this in advance by running the [build job](https://staging.ci.brave.com/job/brave-browser-linux-sign-upload/build?delay=0sec) with the `S3_TEST_BUCKETS` option enabled. This would sign the RPM/DEB and upload to the test S3 repository which you could look at afterwards to see if it was signed as expected, rather than the production S3 bucket.